parser.go

```diff
@@ -2451,6 +2451,15 @@ func (p *Parser) parseCreateIndex(pos token.Pos) *ast.CreateIndex {
 	}
 
 	p.expectKeywordLike("INDEX")
+
+	ifNotExists := false
+	if p.Token.IsKeywordLike("IF") {
+		p.nextToken()
+		p.expect("NOT")
+		p.expect("EXISTS")
+		ifNotExists = true
+	}
```

Comment:
This part would appears every time when parsing IF NOT EXISTS option. So it might be better to extract to a function like `func (*Parser) parseIfNotExists() bool`. What do you think?

Reply:
It sounds good. `IF EXISTS` also appears twice, so we can extract this too. I think a better place to put them would be behind the DDL section.

Reply:
Thanks for suggestion. Same here.
Done. 10becb3